includes/validation/class-amp-validated-url-post-type.php

```diff
@@ -865,16 +865,79 @@ public static function store_validation_errors( $validation_errors, $url, $args
 		return $post_id;
 	}
 
+	/**
+	 * Get recent validation errors by source.
+	 *
+	 * @since 1.6
+	 * @todo This can be stored in object cache, invalidated whenever a validated URL post is inserted/updated/deleted.
+	 *
+	 * @param int $count Maximum count of validated URLs to gather validation errors from.
+	 * @return array Multidimensional array where root keys are source types, sub-keys are source names, and leaf arrays are the validation error terms & data.
+	 */
+	public static function get_recent_validation_errors_by_source( $count = 100 ) {
+		$posts = get_posts(
+			[
+				'post_type'      => self::POST_TYPE_SLUG,
+				'posts_per_page' => $count,
+				'orderby'        => 'date',
+				'order'          => 'desc',
+			]
+		);
+
+		$errors_by_source = [];
+
+		foreach ( $posts as $post ) {
+			// Skip validated URLs which are stale since results will be misleading.
+			if ( self::get_post_staleness( $post ) ) {
+				continue;
+			}
+
+			$validation_errors = self::get_invalid_url_validation_errors( $post );
+			if ( empty( $validation_errors ) ) {
+				continue;
+			}
+
+			foreach ( $validation_errors as $validation_error ) {
+				if ( empty( $validation_error['data']['sources'] ) || ! $validation_error['term'] instanceof WP_Term ) {
+					continue;
+				}
+				foreach ( $validation_error['data']['sources'] as $source ) {
+					if ( ! isset( $source['type'], $source['name'] ) || ! $validation_error['term'] instanceof WP_Term ) {
```

Comment:
Does `$validation_error['term']` needs to be checked again if it's an instance of `WP_Term`?

```suggestion
					if ( ! isset( $source['type'], $source['name'] ) {
```